src/Illuminate/Database/Eloquent/Factories/RealTimeFactory.php

```diff
@@ -0,0 +1,474 @@
+        $this->registerTypeMappings($connection->getDoctrineConnection()->getDatabasePlatform());
+
+        return $this;
+    }
+
+    /**
+     * Set the model for the factory.
+     */
+    public function forModel(string $model): self
+    {
+        $this->model = $model;
+
+        return $this;
+    }
+
+    /**
+     * Create a new instance of the factory builder with the given mutated properties.
+     *
+     * @return static
+     */
+    protected function newInstance(array $arguments = [])
+    {
+        return parent::newInstance($arguments)
+            ->forModel($this->model)
+            ->configure();
+    }
+
+    /**
+     * Get the table columns from the model.
+     */
+    protected function getColumnsFromModel(): Collection
+    {
+        $columns = $this->schema->listTableColumns($this->table);
+
+        return collect($columns);
+    }
+
+    /**
+     * Generate a value for the given column.
+     *
+     * @return void
+     */
+    protected function value(Column $column): mixed
+    {
+        if ($column->getAutoincrement()) {
+            return null;
+        }
+
+        if ($this->isForeignKey($column)) {
+            return null;
+        }
+
+        if ($value = $this->guessValue($column->getName())) {
+            return $value;
+        }
+
+        return ($value = $this->valueFromCast($column->getName())) ?
+            $value :
+            $this->valueFromColumn($column);
+    }
+
+    /**
+     * Determine whether the given column is a foreign key.
+     */
+    protected function isForeignKey(Column $column): bool
+    {
+        return collect($this->schema->listTableForeignKeys($this->table))
+            ->filter(fn ($foreignKey) => in_array($column->getName(), $foreignKey->getLocalColumns()))
+            ->isNotEmpty();
+    }
+
+    /**
+     * Generate a value for the given column.
+     */
+    protected function valueFromColumn(Column $column): mixed
+    {
+        if (! $column->getNotnull() && $column->getDefault() === null) {
+            return null;
+        }
+
+        if ($value = $column->getDefault()) {
+            return $value;
+        }
+
+        return match (true) {
+            $this->isIntegerType($column->getType()) => $this->integerValue(),
+            $this->isDateType($column->getType()) => $this->dateValue(),
+            $this->isDecimalType($column->getType()) => $this->decimalValue($column->getPrecision() ?: 10, $column->getScale() ?: 2),
+            $column->getType() instanceof TimeType => fake()->time(),
+            $column->getType() instanceof BlobType,$column->getType() instanceof TextType => fake()->text(),
+            $column->getType() instanceof BooleanType => $this->booleanValue(),
+            $column->getType() instanceof JsonType,$column->getType() instanceof ArrayType => $this->jsonValue(),
+            default => $this->stringValue(),
```

Comment:
Might be helpful to get the length of the column so that faker has a maximum string length. It's incredibly frustrating to have tests fail intermittently because faker chose a string longer than the width of the column.

Reply:
Good idea, will look into this.

Reply:
Turns out this information isn't accessible via dbal so it's not possible to do it.

Reply:
Turns out you can get it from some drivers so will get this updated.